oauthproxy.go

```diff
@@ -376,6 +389,44 @@ func buildRoutesAllowlist(opts *options.Options) ([]allowedRoute, error) {
 	return routes, nil
 }
 
+// buildHeadersAllowlist builds an []allowedHeader list from the SkipAuthHeaders and LegacySkipAuthHeaders options
+func buildHeadersAllowlist(opts *options.Options) ([]allowedHeader, error) {
```

Comment:
codeclimate complains about too many return statements. Should I move `appendAllowedHeader()` out of here (make it non-anonymous)?